Attribute/AttributeProviderInterface.php

```diff
@@ -11,5 +11,5 @@ interface AttributeProviderInterface
     public function getAttributeByName(
         string $name,
         SpecializationContextHashInterface $contextHash
-    ): ?AttributeInterface;
+    ): AttributeInterface;
```

Comment:
What happens if the attribute doesn’t exist now? @calumbrodie  this change hasn’t introduced an exception or how to handle that?

Reply:
You might also find this is fairly breaky, as this code has been null able for years

Reply:
These providers always return a generic attribute anyway so it makes consuming code not have to check for null returns